utils-intervalrdd/src/main/scala/org/bdgenomics/utils/interval/array/IntervalArray.scala

```diff
@@ -108,6 +108,7 @@ object IntervalArray extends Serializable {
 trait IntervalArray[K <: Interval[K], T] extends Serializable {
   val array: Array[(K, T)]
   val maxIntervalWidth: Long
+  var optLastIndex: Option[Int] = None
```

Comment:
Using `null` here might be an issue because it would require a refactor of the `binarySearch` return type for the change to be impactful. Is that also what you're implying should be changed? I'll do the profiling regardless.